app/models/visit.rb

```diff
@@ -76,4 +76,8 @@ def banned_visitors
   def unlisted_visitors
     visitors.unlisted
   end
+
+  def unavailable?
```

Comment:
In that case, I'm more confused about what this method is for! As I understand it, the scenarios are:

1. The booking is in the initial requested state, and a prison can process it
2. The booking has been processed (accepted, rejected) or withdrawn or cancelled, and they can't do anything except look at it.